packages/scaffold-config/src/ct-detect-third-party.ts

```diff
@@ -0,0 +1,100 @@
+import path from 'path'
+import { pathToFileURL } from 'url'
+import globby from 'globby'
+import { z } from 'zod'
+import Debug from 'debug'
+
+const debug = Debug('cypress:scaffold-config:ct-detect-thrid-party')
+
+const DependencySchema = z.object({
+  type: z.string(),
+  name: z.string(),
+  package: z.string(),
+  installer: z.string(),
+  description: z.string(),
+  minVersion: z.string(),
+})
+
+const DependencyArraySchema = z.array(DependencySchema)
+
+const BundlerSchema = z.enum(['webpack', 'vite'])
+
+const ThirdPartyComponentFrameworkSchema = z.object({
+  type: z.string().startsWith('cypress-ct-'),
+  configFramework: z.string().startsWith('cypress-ct-'),
+  category: z.enum(['library', 'template']),
+  name: z.string(),
+  supportedBundlers: z.array(BundlerSchema),
+  detectors: DependencyArraySchema,
+  dependencies: z.function(),
+  mountModule: z.function(),
+  componentIndexHtml: z.optional(z.function()),
+})
+
+const CT_FRAMEWORK_GLOB = 'node_modules/cypress-ct-*/package.json'
+
+// tsc will compile `import(...)` calls to require unless a different tsconfig.module value
+// is used (e.g. module=node16). To change this, we would also have to change the ts-node behavior when requiring the
+// Cypress config file. This hack for keeping dynamic imports from being converted works across all
+// of our supported node versions
+const _dynamicImport = new Function('specifier', 'return import(specifier)')
+
+const dynamicImport = <T>(module: string) => {
+  return _dynamicImport(module) as Promise<T>
+}
+
+const dynamicAbsoluteImport = (filePath: string) => {
+  return dynamicImport(pathToFileURL(filePath).href) as Promise<any>
+}
+
+export async function detectThirdPartyCTFrameworks (
+  projectRoot: string,
+): Promise<any[]> {
+  try {
+    const fullPathGlob = path.join(projectRoot, CT_FRAMEWORK_GLOB)
+
+    const packageJsonPaths = await globby(fullPathGlob)
+
+    debug('Found packages matching "cypress-ct-*/package.json": %o', packageJsonPaths)
+
+    const modules = await Promise.all(
+      packageJsonPaths.map(async (packageJsonPath) => {
+        try {
+          const packageRoot = path.dirname(packageJsonPath)
+
+          debug('Attempting to resolve third party module with require.resolve: %s', packageRoot)
+
+          const modulePath = require.resolve(packageRoot)
```

Comment:
Importing via node's default import strategy rather than using the `cypress.framework.js` convention. Can refactor depending on where we land.

Reply:
Yeah, we need to sort this out.